src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -294,11 +293,10 @@ void mutex_init() {
   MUTEX_DEFN(CodeHeapStateAnalytics_lock     , PaddedMutex  , safepoint);
   MUTEX_DEFN(ThreadsSMRDelete_lock           , PaddedMonitor, service-2); // Holds ConcurrentHashTableResize_lock
   MUTEX_DEFN(ThreadIdTableCreate_lock        , PaddedMutex  , safepoint);
-  MUTEX_DEFN(SharedDecoder_lock              , PaddedMutex  , service-5); // Must be lower than NmtVirtualMemory_lock due to MemTracker::print_containing_region
+  MUTEX_DEFN(SharedDecoder_lock              , PaddedMutex  , service-5);
```

Comment:
This was originally of rank `tty-1` before introducing `NmtVirtualMemory_lock`. But it was lowered due to rank errors. It probably makes sense to keep this lower rank now, since the actual locking locations have not changed even though `NmtVirtualMemory_lock` is removed. But maybe there should be a comment explaining that this low rank is needed for lock ordering with NMT `MemTracker::print_containing_region`